pkg/model/imagereference.proto

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package pipe.model;
+option go_package = "github.com/pipe-cd/pipe/pkg/model";
+
+import "validate/validate.proto";
+
+message ImageReference {
+    // The generated unique identifier.
+    string id = 1 [(validate.rules).string.min_len = 1];
+    // The repository name.
+    string repo_name = 2 [(validate.rules).string.min_len = 1];
+    // The image digest.
+    string digest = 3;
+    // The image tag.
+    string tag = 4 [(validate.rules).string.min_len = 1];
```

Comment:
For our initial use case, a string is enough, but technically, multiple tags can be associated with a single image reference. For future use, how about making it a comma-separeted list?

Reply:
On second thought, we currently allow the digest to be not unique. So we can keep it string as-is, and deal with it as other references. Performing `docker images` can represent my mind well.
```
$ docker images
REPOSITORY                                                         TAG                 IMAGE ID            CREATED             SIZE
nakabonne-public-test                                              0.1.0               0a0da5a852e7        2 weeks ago         194MB
nakabonne-public-test                                              latest              0a0da5a852e7        2 weeks ago         194MB
```

(Those IMAGE ID is the same but represents as another image reference)

Reply:
In the future, if we want references with the same digest, we just need to perform List(digest=0a0da5a852e7). Then we will have:
- nakabonne-public-test:0.1.0
- nakabonne-public-test:latest
- gcr.io/nakabonne/nakabonne-test:0.1.0

Reply:
Using `array` or `comma-separeted string`  can lead to difficulty for our query at API.
Especially in our case, we support multiple DB backends.
So I want it to be as simple as possible.
The user can store multiple tags on separate entities if they want.

Reply:
Good, let's do so.

Reply:
After rethinking about this, I realized that ImageWatcher only needs to use "repo_name", not "tag" in the query.
And one more reason, by storing them in the same entity we can avoid using transaction/batched-write while adding multiple entities at the same time.
So let me change this too.

Reply:
I see. Yeah, it definitely the best design I think.